packages/react-storage/src/components/StorageBrowser/views/utils.ts

```diff
@@ -34,3 +34,14 @@ export const compareDates = (a: Date, b: Date): number => {
   if (b === undefined) return -1;
   return a.getTime() - b.getTime();
 };
+
+export function isLastPage(
+  currentPage: number,
+  totalItems: number,
+  pageSize: number
+): boolean {
+  // Use Math.ceil so we can round up. For example, if you have
+  // 4 results, and your page size is 3, the last page (Math.ceil(4/3)) will be 2.
+  // If you have 3 results and your page size is 4, the last page will be 1
+  return Math.ceil(totalItems / pageSize) === currentPage;
```

Comment:
Originally [here](https://github.com/aws-amplify/amplify-ui/pull/5936/files#diff-a3e2d76feb6d82570b410f5c6240cc266e736376ca27fc47fad09b946079541cL28).

Reply:
Thanks for adding this back with the comments